### DISCUSSION TITLE
[DRL on Airsim environment : DO WE NEED Test environment also?]

### DISCUSSION THREAD

**Discussion by Charles-Lim93** [microsoft/AirSim#4832] [Support Q&A]
Dear Airsim masters

I'm doing Deep Reinforcement Learning on the Airsim environment, here's the issue.

I ran sample RL in Airsim, **(https://github.com/Microsoft/AirSim/tree/main/PythonClient/reinforcement_learning)**.

And thankfully It worked well for training, but the original code do not have an evaluation code.

So I put  
`model = DQN.load("dqn_airsim_policy", env = env) mean_reward, std_reward = evaluate_policy(model, model.get_env(), n_eval_episodes=5) 
print(mean_reward, std_reward)
`

for example, there are 7 action spaces in a discrete environment, and there was no problem during training.
But when evaluating policy, the action comes out like episode 1's action => 1/2/2 episode 2's action => 1/2/2.... episode 3's action> 1/2/2..... and so on.

Is there anyone who has the similiar issue? Can anyone share their experience on DRL on Airsim?

Thank you in advance.

My environments
- Stable-baseline3
- Airsim 1.6
- Python 3.7
- GTX 2080 Ti

**Comment by Charles-Lim93** [comment]
Update:

I found out that it is due to lack of not enough training set. Thanks all;

**Reply by Gavin2417** [reply]
hi, did you manage to get `env = VecTransposeImage(env)` working, which is a part of the dqn_car.py?
